classification/keras/code/keras_script.py

```diff
@@ -0,0 +1,267 @@
+
+# (Neptune) Track datasets using Neptune
+# Since this dataset will be used among all the runs in the project, we track it at the project level
+
+project["data/keras/files"].track_files(
+    "s3://neptune-examples/data/text-classification/aclImdb_v1.tar.gz"
+)
+project.sync()
+
+# (Neptune) Download files from S3 using Neptune
+
+print("Downloading data...")
+project["data/keras/files"].download("..")
+
+# Prepare data
+
+utils.extract_files(source="../aclImdb_v1.tar.gz", destination="..")
+utils.prep_data(imdb_folder="../aclImdb", dest_path="../data")
+
+# (Neptune) Upload dataset sample to Neptune project
+
+base_namespace = "data/keras/sample/"
+
+project[base_namespace]["train/pos"].upload(
+    f"../data/train/pos/{random.choice(os.listdir('../data/train/pos'))}"
+)
+project[base_namespace]["train/neg"].upload(
+    f"../data/train/neg/{random.choice(os.listdir('../data/train/neg'))}"
+)
+project[base_namespace]["test/pos"].upload(
+    f"../data/test/pos/{random.choice(os.listdir('../data/test/pos'))}"
+)
+project[base_namespace]["test/neg"].upload(
+    f"../data/test/neg/{random.choice(os.listdir('../data/test/neg'))}"
+)
+
+# Generate training, validation, and test datasets
+
+data_params = {
+    "batch_size": 32,
+    "validation_split": 0.2,
+    "max_features": 2000,
+    "embedding_dim": 128,
+    "sequence_length": 500,
+    "seed": 1,
+}
+
+# (Neptune) Log data metadata to Neptune
+
+run = neptune.init_run(name="Keras text classification", tags=["keras"])
+
+run["data/params"] = data_params
+
+raw_train_ds, raw_val_ds = tf.keras.preprocessing.text_dataset_from_directory(
+    "../data/train",
+    batch_size=data_params["batch_size"],
+    validation_split=data_params["validation_split"],
+    subset="both",
+    seed=data_params["seed"],
+)
+
+raw_test_ds = tf.keras.preprocessing.text_dataset_from_directory(
+    "../data/test", batch_size=data_params["batch_size"]
+)
+
+print(f"Number of batches in raw_train_ds: {raw_train_ds.cardinality()}")
+print(f"Number of batches in raw_val_ds: {raw_val_ds.cardinality()}")
+print(f"Number of batches in raw_test_ds: {raw_test_ds.cardinality()}")
+
+
+# Clean data
+
+
+def custom_standardization(input_data):
+    lowercase = tf.strings.lower(input_data)
+    stripped_html = tf.strings.regex_replace(lowercase, "<br />", " ")
+    return tf.strings.regex_replace(stripped_html, f"[{re.escape(string.punctuation)}]", "")
+
+
+vectorize_layer = TextVectorization(
+    standardize=custom_standardization,
+    max_tokens=data_params["max_features"],
+    output_mode="int",
+    output_sequence_length=data_params["sequence_length"],
+)
+
+text_ds = raw_train_ds.map(lambda x, y: x)
+vectorize_layer.adapt(text_ds)
+
+# Vectorize data
+
+
```

Comment:
Blank line between comment and function.

Reply:
Black does this to prevent flake8 from raising an E302 error

Reply:
Maybe put the comment in the body of the function or maybe just remove it?